modules/post/windows/gather/checkvm.rb

```diff
@@ -180,6 +208,59 @@ def qemu?
     key_path = 'HKLM\HARDWARE\DESCRIPTION\System\CentralProcessor\0'
     return true if registry_getvaldata(key_path, 'ProcessorNameString') =~ /qemu/i
 
+    return true if registry_getvaldata('HKLM\HARDWARE\DESCRIPTION\System', 'SystemBiosVersion') =~ /qemu/i
+    return true if registry_getvaldata('HKLM\HARDWARE\DESCRIPTION\System', 'VideoBiosVersion') =~ /qemu/i
+    return true if registry_getvaldata('HKLM\HARDWARE\DESCRIPTION\System\BIOS', 'SystemManufacturer') =~ /qemu/i
+
+    false
+  end
+
+  def joesandbox?
+    vpcprocs = [
+      'joeboxcontrol.exe',
+      'joeboxserver.exe'
+    ]
+    get_processes.each do |x|
+      vpcprocs.each do |p|
+        return true if p == x['name'].downcase
+      end
+    end
+
+    key_path = 'HKLM\SOFTWARE\Microsoft\Windows\CurrentVersion'
+    return true if registry_getvaldata(key_path, 'ProductId') == '55274-640-2673064-23950'
+    key_path = 'HKLM\SOFTWARE\Microsoft\Windows NT\CurrentVersion'
+    return true if registry_getvaldata(key_path, 'ProductId') == '55274-640-2673064-23950'
+
+    false
+  end
+
+  def microsoftsandbox?
+    # also known as Wundows Containers
```

Comment:
Windows or windows?

Reply:
[Woensel]( https://www.youtube.com/watch?v=hIKP7dFSG9E ) of course.